### PR TITLE
Changes for DMP test suite

### PR DESCRIPTION
**Pull request by VolodymyrLavrenchuk**

### CODE REVIEW

File: http_calls.py [hunk 1]
```diff
@@ -71,8 +74,11 @@ def read(self, req):
             ret = resp.read()
             print("Response body: %s" % ret)
             global lastRequestResult
-
-            lastRequestResult = ret.decode('utf-8')
+            try:
+              lastRequestResult = ret.decode('utf-8')
+            except BaseException as e:
+              print("Can`t decode utf-8. Try to use koi8-r")
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py]: Test: FilesEntity
download for binary file returns:
b'"\xf3\xc9\xcd\xd7\xcf\xcc\xd9 \xd7 \xcb\xcf\xc4\xc9\xd2\xcf\xd7\xcb\xc5 KOI-8R"\n'

File: http_calls.py [hunk 2]
```diff
@@ -392,7 +398,7 @@ def get_dataset(self):
                                 if isinstance(val, list) and k.isdigit():
                                     k = int(k)
                                 elif isinstance(val, dict) and k.isdigit():
-                                    k = val.keys()[int(k)]
+                                    k = list(val.keys())[int(k)]
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py:401]: In Python3 dict.keys() returns dict not list

File: http_calls.py [hunk 3]
```diff
@@ -128,9 +134,9 @@ def get_str(self, url, args=None):
         return self.GET(self.get_full_url(url), {'Accept': 'application/json'}, args).decode('utf-8')
 
     def get_hex_str(self, url, args=None):
-        resp = self.get_str(url, args)
+        resp = self.GET(self.get_full_url(url), {'Accept': 'application/json'}, args)
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py:137]: For test FilesEntity:
need to do "hexlify" before "decode('utf-8')":
'utf-8' codec can't decode byte 0xf3 in position 1: invalid continuation byte

File: http_calls.py [hunk 4]
```diff
@@ -428,7 +434,10 @@ def __init__(self):
         global lastRequestResult
 
         o = json.loads(lastRequestResult)
-        self.result = o['hits']['hits']
+        if(type(o) == dict and "hits" in o):
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py]: Test: ObjectEntity/GetMultipleInstances
Request: /api/objects/multiple returns items in 'doc' key but not in 'hits':
b'{"docs":[{..}]'

**Review by VolodymyrLavrenchuk — COMMENTED** [review]
Describe changes & tests that have been fixed

File: http_calls.py [hunk 5]
```diff
@@ -446,10 +455,7 @@ def __init__(self, url, args=None):
             body = body[g_array_field]
 
         global lastResponse
-        print(lastResponse.info())
-
-        self.result = {'status_code': lastResponse.getcode(), 'headers': lastResponse.info().dict, 'body': body}
-
+        self.result = {'status_code': lastResponse.getcode(), 'headers': lastResponse.info(), 'body': body}
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py]: Test: LockingApi.LeaseCredentials:
http.client.HTTPMessage in "info()" does not have dict method

File: http_calls.py [hunk 6]
```diff
@@ -447,9 +456,7 @@ def __init__(self, url, args=None):
 
         global lastResponse
         print(lastResponse.info())
-
-        self.result = {'status_code': lastResponse.getcode(), 'headers': lastResponse.info().dict, 'body': body}
-
+        self.result = {'status_code': lastResponse.getcode(), 'headers': lastResponse.info(), 'body': body}
```
Comment (VolodymyrLavrenchuk) [inline http_calls.py:460]: Test: LockingApi.LeaseCredentials:
http.client.HTTPMessage in "info()" does not have dict method

File: http_calls.py (hunk 1 above)
Comment (grigoryvasiliev) [reply http_calls.py]: what is the use case of non utf-8 result?

Comment (VolodymyrLavrenchuk) [reply http_calls.py]: When we want to post some binary data, for ex. binary file

Comment (grigoryvasiliev) [reply http_calls.py]: what is real use case?

Comment (grigoryvasiliev) [reply http_calls.py]: koi8 is not binary - right?

File: http_calls.py [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import ast
 import re
 import time
+import codecs
```
Comment (AntonTimiskov) [inline http_calls.py]: not used

**Pull request merged**